DotnetFoundation/DotnetFoundation.Api/Controllers/TaskController.cs

```diff
@@ -20,104 +20,180 @@ public TaskController(ITaskDetailsService TaskDetailsService)
     /// <summary>
     /// Get all tasks.
     /// </summary>
-    [HttpGet]    
+    [HttpGet("tasks")]
     [ProducesResponseType(StatusCodes.Status200OK)]
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
-    public async Task<ActionResult<BaseResponse<List<TaskDetailsResponse>>>> GetAllTasksAsync()
+    public async Task<ActionResult<BaseResponse<PagedList<TaskDetailsResponse>>>> GetAllTasksAsync([FromQuery] PagingRequest pagingRequest)
     {
-        BaseResponse<List<TaskDetailsResponse>> response = new(ResponseStatus.Fail);
-
-        response.Data = await _taskDetailsService.GetAllTasksAsync().ConfigureAwait(false);
-        response.Status = ResponseStatus.Success;
-
-        return Ok(response);
+        BaseResponse<PagedList<TaskDetailsResponse>> response = new(ResponseStatus.Fail);
```

Comment:
What is the meaning on `new(ResponseStatus.Fail)`. It should be the instance of BaseResponse.

Reply:
It is creating a BaseResponse<T> Object `response` with ResponseStatus of enum Fail